differ/differ.go

```diff
@@ -114,27 +113,22 @@ func NewFileAttribute(fileName string) *FileAttributes {
 
 type oneEntry struct {
 	Key               string
+	CrMeta            *crMeta.CRMetadata
+	pRev              uint64
```

Comment:
What's `pRev` doing here?
Shouldn't `pRev` be used in as part of `UpdateHLVIfNeeded` to determine if the HLV needs to be updated?

It looks out of place... given `pRev's` purpose is as per the function states:

```
	if importCas == cas {
		// This is an import mutation. cvCas represents the pre-import CAS and that is used for CR
		// pRev is the pre-import revId which is used for CR
		m.isImport = true
		m.docMeta.Cas = cvCas
		m.docMeta.RevSeq = pRev
	}
```

Wouldn't that suffice?
The idea is that the pRev/pCAS/importCAS are all used to decide at the time `CRMetadata` is constructed, the `CRMetadata.docMeta.Cas` and `CRMetadata.docMeta.RevSeq` should be and the diff function should take care of the rest?

i.e. 
```(source *CRMetadata) Diff(target *CRMetadata,```


If not, can you explain to me the thought logic behind this?

Reply:
I am actually mocking UpdateHLVIfNeeded.
That's the reason pRev field and SetVersion functions exist.
Reason/Thought Process : crMeta.UpdateHLVIfNeeded function has the following signature.
`func (m *CRMetadata) UpdateHLVIfNeeded(source hlv.DocumentSourceId, cas uint64, cvCas uint64, cvSrc hlv.DocumentSourceId, cvVer uint64, pvMap hlv.VersionsMap, mvMap hlv.VersionsMap, importCas uint64, pRev uint64) (err error)`

If this function is used to set the right versions of the document for comparison then it expects you to have HLV fields to be passed.
It works under the assumption that if importCas exists then HLV's will always be present even if EnableCrossClusterVersioning is set to false.
I had to do an assumption which i was not sure of and hence i decided to do version updation and HLV creation as two seperate things.

We could proceed to use **UpdateHLVIfNeeded** if importCas is present implies HLV is present irrespective of the bucket setting.